<launch>
  <arg name="robot" default="scitos"/>
  <!-- views: DEP or IND or IND--> 
  <arg name="setting" default="IND"/>  
  <!-- algorithm: SRC, DRC, SR, DR, SC, DC --> 
  <arg name="alg" default="SRC"/> 
 
  <arg name="num_of_plans" default="2000"/>
  <!-- <arg name="plan_length" default="25"/> -->
  <arg name="time_window" default="10"/>
  <arg name="rho" default="1.0"/>
  <arg name="input_file" default="$(find viper)/data/op/$(arg robot)/view_keys.json"/>
  <arg name="input_file_values" default="$(find viper)/data/op/$(arg robot)/view_values.json"/>
  <arg name="input_file_costs" default="$(find viper)/data/op/$(arg robot)/view_costs.json"/>
b

  <!-- <arg name="run" default="123"/>-->
  <arg name="output_file" default="$(find viper)/data/op/$(arg robot)/plans/View$(arg setting)-Alg$(arg alg)-N$(arg num_of_plans)-T$(arg time_window)-Rho$(arg rho).json"/>

  <arg name="output_file_best" default="$(find viper)/data/op/$(arg robot)/plans/View$(arg setting)-Alg$(arg alg)-N$(arg num_of_plans)-T$(arg time_window)-Rho$(arg rho)-BEST.json"/>


  <include file="$(find viper)/launch/$(arg robot)/_generate_plans.launch"/>

  <node name="generate_plans" pkg="viper" type="gen_plans.py" output="screen">
    <param name="robot" type="string" value="$(arg robot)"/>
    <param name="num_of_plans" type="int" value="$(arg num_of_plans)"/>
    <!-- <param name="plan_length" type="int" value="$(arg plan_length)"/>-->
    <param name="time_window" type="int" value="$(arg time_window)"/>
    <param name="rho"  type="double" value="$(arg rho)"/>
    <param name="input_file" type="string" value="$(arg input_file)"/>
    <param name="input_file_values" type="string" value="$(arg input_file_values)"/>
    <param name="input_file_costs"  type="string" value="$(arg input_file_costs)"/>
    <param name="output_file"       type="string" value="$(arg output_file)"/>
    <param name="output_file_best"  type="string" value="$(arg output_file_best)"/>
  </node>
</launch>
